src/main/scala/net/ruippeixotog/scalascraper/browser/Browser.scala

```diff
@@ -15,6 +15,8 @@ trait Browser {
   def parseFile(path: String, charset: String): Document = parseFile(new File(path), charset)
   def parseFile(path: String): Document = parseFile(new File(path), "UTF-8")
 
+  def parseResource(name: String, charset: String = "UTF-8"): Document = parseFile(new File(getClass.getResource(name).toURI), charset)
```

Comment:
You are, of course, completely right.

The 'problem' is converting from file to string is not as straightforward as it sounds: the obvious approach of `Source.fromInputStream(is)(charset).mkString` will throw an exception when the input stream contains characters that are not in the specified charset.

JSoup has some pretty intricate mechanisms to deal with this situation more gracefully (https://github.com/jhy/jsoup/blob/master/src/main/java/org/jsoup/helper/DataUtil.java#L91)

Perhaps the reasonable middle ground would be to:
* detect whether the resource is from the filesystem
* if so rely on the `Browser` to parse the file
* otherwise use `Source.fromInputStream` (and accept the fact that it doesn't allow incorrectly-encoded input)

What do you think?